src/python/pants/BUILD

```diff
@@ -27,6 +27,8 @@ python_distribution(
         # consumers of pantsbuild.pants are using a compatible interpreter.
         # TODO(7344): the tuple syntax for ext_modules is deprecated. Use Extension once we support it.
         ext_modules=[("native_engine", {"sources": ["pants/dummy.c"]})],
+        # N.B.: Must match [python] interpreter_constraints in pants.toml.
+        python_requires=">=3.7,<3.10",
```

Comment:
Hmm.. this duplication is a bit sad tbh.

Reply:
A plugin that reads pants.toml (or uses Config) to get the unadulterated ICs and automatically apply them. We used to have a custom ~python_distribution in-repo plugin for just this sort of boilerplate. It may have only been for the JVM jars we published.